esptfa_arima/Authentication/services.py

```
@@ -31,11 +38,161 @@ def register_user(username, password, first_name, last_name, email, acc_type, lr
+            try:
+                lrn = str(row["lrn"]).strip()
+                if len(lrn) != 11:
+                    raise DRFValidationError(f"Row {index+1}:LRN '{lrn}' is not 11 characters long")
+                first_name = str(row["first_name"]).strip()
+                middle_name = str(row["middle_name"]).strip() if pd.notna(row["middle_name"]) else ""
+                last_name = str(row["last_name"]).strip()
+
+                # see if student already exists
+                student_obj = Student.objects.filter(lrn=lrn, section=section).first()
+                if student_obj:
+                    raise DRFValidationError(f"Row {index+1}:LRN '{lrn}' already exists in section '{section_name}'")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)